var/spack/repos/builtin/packages/lbann/package.py

```diff
@@ -347,6 +347,7 @@ def cache_name(self):
 
     def initconfig_compiler_entries(self):
         spec = self.spec
+        spec.compiler_flags["cxxflags"].append("-std=c++17")
```

Comment:
I'm surprised this was necessary in addition to setting the CMAKE_CXX_STANDARD below. Would you be able to add the error you get without this? I would also recommend consolidating this with `setup_build_environment` which uses `env.append_flags` for a couple reasons:

* Modifying this data structure is not idiomatic (i.e. not how we normally effect changes of this sort
* If you kept this here, you'd have two different mechanisms for modifying flags (which, given that you are the maintainer of this, is more of a suggestion than a requirement)

Reply:
It is no longer necessary.